drivers/sensor/kx022/kx022_trigger.c

```diff
@@ -0,0 +1,315 @@
+}
+static void kx022_handle_slope_int(const struct device *dev)
+{
+	struct kx022_data *data = dev->data;
+
+	if (data->slope_handler != NULL) {
+		data->slope_handler(dev, &data->slope_trigger);
+	}
+}
+
+static void kx022_handle_int(const struct device *dev)
+{
+	struct kx022_data *data = dev->data;
+	const struct kx022_config *cfg = dev->config;
+	uint8_t status, clr;
+
+	if (data->hw_tf->read_reg(data, KX022_REG_INS2, &status) < 0) {
+		return;
+	}
+
+	if (status & 0x02) {
+		printf("===================MOTION DETECTED===============\r\n");
+		kx022_handle_motion_int(dev);
+	} else if (status & 0x10) {
+		printf("======================DATA READY==================\r\n");
+
+		kx022_handle_drdy_int(dev);
+		k_msleep(50);
+
+	} else if (status & 0x01) {
+		printf("==============TILT POSITION DETECTED============\r\n");
+		kx022_handle_slope_int(dev);
+	}
+	data->hw_tf->read_reg(data, KX022_REG_INT_REL, &clr);
+	gpio_pin_interrupt_configure(data->gpio, cfg->irq_pin, GPIO_INT_EDGE_TO_ACTIVE);
+}
+
+#ifdef CONFIG_KX022_TRIGGER_OWN_THREAD
+static void kx022_thread(struct kx022_data *data)
+{
+	while (1) {
+		k_sem_take(&data->trig_sem, K_FOREVER);
+		kx022_handle_int(data->dev);
+	}
+}
+#endif
+
+#ifdef CONFIG_KX022_TRIGGER_GLOBAL_THREAD
+static void kx022_work_cb(struct k_work *work)
+{
+	struct kx022_data *data = CONTAINER_OF(work, struct kx022_data, work);
+
+	kx022_handle_int(data->dev);
+}
+#endif
+
+/************************************************************
+ * info: kx022_mode use to set pc1 in standby or operating mode
+ * input: 0 - standby
+ *        1 - operating
+ * **********************************************************/
+
+void kx022_mode(const struct device *dev, bool mode)
+{
+	struct kx022_data *data = dev->data;
+
+	data->hw_tf->update_reg(data, KX022_REG_CNTL1, KX022_MASK_CNTL1_PC1,
+				(mode << KX022_CNTL1_PC1_SHIFF));
+}
+/**************************************************************
+ * FUNCTION : kx022_trigger_init
+ * use to initialize trigger function
+ *
+ * ************************************************************/
+int kx022_trigger_init(const struct device *dev)
+{
+	struct kx022_data *data = dev->data;
+	const struct kx022_config *cfg = dev->config;
+
+	/* setup data ready gpio interrupt */
+	data->gpio = device_get_binding(cfg->irq_port);
+	if (data->gpio == NULL) {
+		LOG_ERR("Cannot get pointer to %s device.", cfg->irq_port);
+		return -EINVAL;
+	}
+
+	gpio_pin_configure(data->gpio, cfg->irq_pin, GPIO_INPUT | cfg->irq_flags);
+
+	gpio_init_callback(&data->gpio_cb, kx022_gpio_callback, BIT(cfg->irq_pin));
+
+	if (gpio_add_callback(data->gpio, &data->gpio_cb) < 0) {
+		LOG_ERR("Could not set gpio callback.");
+		return -EIO;
+	}
+	data->dev = dev;
+	switch (KX022_DEFAULT_INT_PIN) {
+	case 1:
+		data->hw_tf->update_reg(data, KX022_REG_INC1, KX022_MAKS_INC1_INT_EN,
+					KX022_INT1_EN);
+		break;
+	case 2:
+		data->hw_tf->update_reg(data, KX022_REG_INC5, KX022_MAKS_INT2_EN, KX022_INT2_EN);
+		break;
+	default:
```

Comment:
default case likely not required here, if there's only 2 possibilities in the Kconfig